### PR TITLE
fix: ignore spurious bindingCalled events

### PR DESCRIPTION
**Pull request by johanbay**

### CODE REVIEW

File: src/common/DOMWorld.ts [hunk 1]
```diff
@@ -537,7 +537,15 @@ export class DOMWorld {
   private async _onBindingCalled(
     event: Protocol.Runtime.BindingCalledEvent
   ): Promise<void> {
-    const { type, name, seq, args } = JSON.parse(event.payload);
+    let payload: { type: string; name: string; seq: number; args: unknown[] };
+    try {
+      payload = JSON.parse(event.payload);
+    } catch {
+      // The binding was either called by something in the page or it was
+      // called before our wrapper was initialized
```
Comment (jackfranklin) [inline src/common/DOMWorld.ts]: just for my understanding: what might cause this? Just some code running before we set up our binding ?

Comment (johanbay) [reply src/common/DOMWorld.ts]: Exactly. I can't make out how it happens, but very rarely we'll call the `ariaQueryHandler` binding before the [pageBinding](https://github.com/puppeteer/puppeteer/blob/main/src/common/helper.ts#L180) wrapper is set up.

**Review by jackfranklin — APPROVED** [review]

File: src/common/DOMWorld.ts (hunk 1 above)
Comment (mathiasbynens) [inline src/common/DOMWorld.ts]: nitty nit

```suggestion
      // called before our wrapper was initialized.
```

File: src/common/Page.ts [hunk 2]
```diff
@@ -1132,7 +1132,15 @@ export class Page extends EventEmitter {
   private async _onBindingCalled(
     event: Protocol.Runtime.BindingCalledEvent
   ): Promise<void> {
-    const { type, name, seq, args } = JSON.parse(event.payload);
+    let payload: { type: string; name: string; seq: number; args: unknown[] };
+    try {
+      payload = JSON.parse(event.payload);
+    } catch {
+      // The binding was either called by something in the page or it was
+      // called before our wrapper was initialized
```
Comment (mathiasbynens) [inline src/common/Page.ts]:
```suggestion
      // called before our wrapper was initialized.
```

**Review by mathiasbynens — APPROVED** [review]

**Pull request merged**